multisrc/overrides/mangathemesia/komikcast/src/KomikCast.kt

```diff
@@ -45,6 +46,16 @@ class KomikCast : MangaThemesia(
         return GET(page.imageUrl!!, newHeaders)
     }
 
+    override fun popularMangaRequest(page: Int) = customPageRequest(page, "orderby", OrderByFilter("popular"))
+    override fun latestUpdatesRequest(page: Int) = customPageRequest(page, "sortby", OrderByFilter("update"))
+
+    private fun customPageRequest(page: Int, filterKey: String, filter: OrderByFilter): Request {
+        val url = baseUrl.toHttpUrl().newBuilder().addPathSegment(mangaUrlDirectory.substring(1))
+        val onPage = if (page > 1) "page/$page/" else ""
+        url.addPathSegments(onPage).addQueryParameter(filterKey, filter.selectedValue())
+        return GET(url.toString())
+    }
```

Comment:
You can simplify the code by using Strings directly here.

Also, there's no need to use `toHttpUrl` as the values are simple and doesn't depend on encoding.

```suggestion
    override fun popularMangaRequest(page: Int) = customPageRequest(page, "orderby", "popular")
    override fun latestUpdatesRequest(page: Int) = customPageRequest(page, "sortby", "update")

    private fun customPageRequest(page: Int, filterKey: String, filterValue: String): Request {
        val pagePath = if (page > 1) "/page/$page/" else ""
        
        return GET("$baseUrl$mangaUrlDirectory/$pagePath?$filterKey=$filterValue", headers)
    }
```